src/mantarray_desktop_app/mc_comm.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+"""Process controlling communication with Mantarray Microcontroller."""
+from __future__ import annotations
+
+import logging
+from multiprocessing import Queue
+from time import sleep
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Tuple
+
+from .constants import SERIAL_COMM_MAGIC_WORD_BYTES
+from .constants import SERIAL_COMM_MAX_PACKET_LENGTH_BYTES
+from .constants import SERIAL_COMM_STATUS_BEACON_PERIOD_SECONDS
+from .exceptions import SerialCommPacketRegistrationReadEmptyError
+from .exceptions import SerialCommPacketRegistrationSearchExhaustedError
+from .exceptions import SerialCommPacketRegistrationTimoutError
+from .instrument_comm import InstrumentCommProcess
+
+
+class McCommunicationProcess(InstrumentCommProcess):
+    """Process that controls communication with the Mantarray Beta 2 Board(s).
+
+    Args:
+        board_queues: A tuple (the max number of MC board connections should be pre-defined, so not a mutable list) of tuples of 3 queues. The first queue is for input/communication from the main thread to this sub process, second queue is for communication from this process back to the main thread. Third queue is for streaming communication (largely fo raw data) to the process that controls writing to disk.
+        fatal_error_reporter: A queue that reports back any unhandled errors that have caused the process to stop.
+        suppress_setup_communication_to_main: if set to true (often during unit tests), messages during the _setup_before_loop will not be put into the queue to communicate back to the main process
+    """
+
+    def __init__(
+        self,
+        # pylint: disable=duplicate-code
+        board_queues: Tuple[  # pylint-disable: duplicate-code
+            Tuple[
+                Queue[Dict[str, Any]],  # pylint: disable=unsubscriptable-object
+                Queue[  # pylint: disable=unsubscriptable-object
+                    Dict[str, Any]
```

Comment:
could this be replaced with `*args, **kwargs` also (like your suggestion for the simulator)?

Reply:
I suppose it could, I can test it out in a new branch

Reply:
Seems to work fine